src/main/java/com/flat/odessa/entity/Links.java

```diff
@@ -0,0 +1,190 @@
+package com.flat.odessa.entity;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class Links implements Iterable<String> {
```

Comment:
@greySlon can we define `Links` interface that extends `Iterable<URL>` and have one implementation that grabs URLs from single page?